src/test/RwaSpell.t.sol

```diff
@@ -0,0 +1,1005 @@
+        hevm.store(
+            address(rwaurn),
+            keccak256(abi.encode(address(this), uint256(1))),
+            bytes32(uint256(1))
+        );
+        assertEq(rwagem.totalSupply(), 1 * WAD);
+        assertEq(rwagem.balanceOf(address(this)), 1 * WAD);
+        assertEq(rwaurn.can(address(this)), 1);
+
+        rwagem.approve(address(rwaurn), 1 * WAD);
+        rwaurn.lock(1 * WAD);
+        rwaurn.draw(1 * WAD);
+
+        assertEq(dai.balanceOf(address(rwaconduitout)), 1 * WAD);
+
+        // wards
+        hevm.store(
+            address(rwaconduitout),
+            keccak256(abi.encode(address(this), uint256(0))),
+            bytes32(uint256(1))
+        );
+
+        // can
+        hevm.store(
+            address(rwaconduitout),
+            keccak256(abi.encode(address(this), uint256(1))),
+            bytes32(uint256(1))
+        );
+
+        assertEq(dai.balanceOf(address(rwaconduitout)), 1 * WAD);
+
+        rwaconduitout.kiss(address(this));
+        rwaconduitout.pick(address(this));
+
+        rwaconduitout.push();
+
+        assertEq(dai.balanceOf(address(this)), 1 * WAD);
+
+        dai.transfer(address(rwaconduitin), dai.balanceOf(address(this)));
+        rwaconduitin.push();
+
+        assertEq(dai.balanceOf(address(rwaurn)), 1 * WAD);
+
+        rwaurn.wipe(1 * WAD);
+        rwaurn.free(1 * WAD);
+    }
+
+    function testSpellIsCast_RWA001_END() public {
+        vote();
+        scheduleWaitAndCast();
+        assertTrue(spell.done());
+
+        endSpell = new EndSpell();
+        voteTemp(address(endSpell));
+
+        endSpell.schedule();
+
+        uint256 castTime = block.timestamp + pause.delay();
+        hevm.warp(castTime);
+        endSpell.cast();
+
+        // TODO: finish
+    }
+
+    // TODO: finish up
+    // function testSpellIsCast_OUTPUT_CONDUIT() public {
+    //     vote();
+    //     scheduleWaitAndCast();
+    //     assertTrue(spell.done());
+
+    //     // fix this
+    //     hevm.store(
+    //         address(rwaconduitout),
+    //         keccak256(abi.encode(address(a), uint256(4))),
+    //         uint256(1)
+    //     );
+    //     hevm.store(
+    //         address(rwaconduitout),
+    //         keccak256(abi.encode(address(b), uint256(4))),
+    //         uint256(1)
+    //     );
+    //     hevm.store(
+    //         address(rwaconduitout),
+    //         keccak256(abi.encode(address(c), uint256(4))),
+    //         uint256(1)
+    //     );
+    //     hevm.store(
+    //         address(rwaurn),
+    //         keccak256(abi.encode(address(this), uint256(1))),
+    //         bytes32(uint256(1))
+    //     );
+
+    // }
+
+    // test end in 2 scenarios (if we need yank in flip contract?) think thru if we need yank in the clip?
+
+    //  what to do if MCD isn't instantiated, if we don't give people the token big issue, token can trade on uni
+
+    // would this end work with ETH? think of it very generally
```

Comment:
All of this is resolved, as the token can be exited from the auth gem join and traded after ES.  That same token could be deposited in the new system.  And we got rid of the Flipper.